usuarios/views.py

```diff
@@ -24,52 +27,103 @@ class CreateView(crud.base.CrudCreateView):
 
 
-class HabilitarDetailView(LoginRequiredMixin, crud.base.CrudDetailView):
-    template_name = "usuarios/habilitar_detail.html"
+class ConveniadoView(PermissionRequiredMixin, FormView):
+    template_name = "crud/form.html"
+    permission_required = {'usuarios.change_usuario',
+                           'usuarios.can_change_conveniado',
+                           'usuarios.add_usuario'}
 
     def get(self, request, *args, **kwargs):
         context = {}
+
+        usuario = Usuario.objects.get(pk=self.kwargs['pk'])
+        form = ConveniadoEditForm(instance=usuario)
+
         context['pk'] = self.kwargs['pk']
-        context['usuario'] = Usuario.objects.get(pk=self.kwargs['pk'])
+        context['form'] = form
+
         return self.render_to_response(context)
 
+    def post(self, request, *args, **kwargs):
+        form = ConveniadoEditForm(request.POST)
+        usuario = Usuario.objects.get(pk=self.kwargs['pk'])
+        if form.data.get('conveniado') == 'on':
+            usuario.conveniado = True
+        usuario.data_ultima_atualizacao = timezone.now()
+        if usuario.conveniado is True and usuario.responsavel is True:
+            usuario.habilitado = True
+            user = User.objects.get(usuario__id=usuario.id)
+            permissao = Permission.objects.get(
+                name='Can add Solicitação de Novo Serviço')
+            user.user_permissions.add(permissao)
+        usuario.save()
+        return self.form_valid(form)
+
+    def get_success_url(self):
+        return reverse('usuarios:usuario_list')
+
 
-class HabilitarEditView(LoginRequiredMixin, FormView):
+class ResponsavelView(PermissionRequiredMixin, FormView):
     template_name = "crud/form.html"
+    permission_required = {'usuarios.change_usuario',
+                           'usuarios.can_change_responsavel',
+                           'usuarios.add_usuario'}
+
+    def handle_no_permission(self):
+        if self.raise_exception:
+            raise PermissionDenied(self.get_permission_denied_message())
+        return HttpResponseRedirect(reverse(
+            'usuarios:conveniado_edit', kwargs={'pk': self.kwargs['pk']}))
 
     def get(self, request, *args, **kwargs):
         context = {}
 
         usuario = Usuario.objects.get(pk=self.kwargs['pk'])
-        form = HabilitarEditForm(instance=usuario)
+        form = ResponsavelEditForm(instance=usuario)
 
         context['pk'] = self.kwargs['pk']
         context['form'] = form
+
         return self.render_to_response(context)
 
     def post(self, request, *args, **kwargs):
-        form = HabilitarEditForm(request.POST)
+        form = ResponsavelEditForm(request.POST)
         usuario = Usuario.objects.get(pk=self.kwargs['pk'])
-        usuario.habilitado = str2bool(form.data['habilitado'])
+        usuario.responsavel = str2bool(form.data['responsavel'])
         usuario.data_ultima_atualizacao = timezone.now()
-
+        if usuario.conveniado is True and usuario.responsavel is True:
```

Comment:
Python idiomático seria ``if usuario.conveniado and usuario.responsavel:``, não @marciomazza ?